Mathlib/Tactic/StacksAttribute.lean

```diff
@@ -43,48 +43,87 @@ It extends the `Tag` environment extension with the data `declName, tag, comment
 def addTagEntry {m : Type → Type} [MonadEnv m] (declName : Name) (tag comment : String) : m Unit :=
   modifyEnv (tagExt.addEntry · { declName := declName, tag := tag, comment := comment })
 
-/--
-The syntax for a Stacks tag: it is an optional number followed by an optional identifier.
-This allows `044Q3` and `GH3F6` as possibilities.
--/
-declare_syntax_cat stackTag
+open Parser
+
+/-- `stacksTag` is the node kind of Stacks Project Tags: a sequence of digits and
+uppercase letters. -/
+abbrev stacksTagKind : SyntaxNodeKind := `stacksTag
+
+/-- The main parser for Stacks Project Tags: it accepts any sequence of 4 digits or
+uppercase letters. -/
+def stacksTagFn : ParserFn := fun c s =>
+  let i := s.pos
+  let s := takeWhileFn (fun c => c.isAlphanum) c s
+  if s.hasError then
+    s
+  else if s.pos == i then
+    ParserState.mkError s "stacks tag"
+  else
+    let tag := Substring.mk c.input i s.pos |>.toString
```

Comment:
I think this causes a caching issue: if I edit the file in vscode, the errors become outdated.

Reply:
@david-christiansen, perhaps you can advise if this is something that is sensible to do in a parser, since you wrote many for Verso!

Reply:
I tried this in VSCode and I was not able to get the caching issue: the parsing seemed to always be correct.

Regardless of whether the saved file was in a "correct" or "incorrect" state, editing (and not saving) would produce the correct parsing outcome.